### PR TITLE
Rewrite on tools to use new structure

### PR DESCRIPTION
**Pull request by Eirmas**
The tools now access a shape creator class to make it easier and less messy to create the shapes. Also did some cleanup in canvasElement object and added interface for returntypes

### CODE REVIEW

File: src/tools/Circle.ts [hunk 1]
```diff
@@ -3,11 +3,12 @@ import Konva from 'konva'
 import { CanvasElement } from '@/types/Canvas'
 import uuid from 'uuid'
 import throttle from 'lodash.throttle'
+import CircleCreator from '@/tools/shapes/CircleCreator'
 
 export default class Circle implements Tool {
   private circle: Konva.Circle;
-  private line: Konva.Line;
-  private stroke: number[][];
+  private line?: Konva.Line;
```
Comment (bryansamuel1) [inline src/tools/Circle.ts]: I am surprised it hasn't complained when using `;` can we remove them?

File: src/tools/Ruler.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    const result = this.rulerCreator.create(canvasElement, layer)
+    const pos = {
+      x: canvasElement.data[2],
+      y: canvasElement.data[3]
+    }
+    this.rulerCreator.move(canvasElement, layer, pos, result.line, result.text, result.circle)
+    layer.batchDraw()
+  }
+
+  sendToWebsockets = (socket: WebSocket, canvasElement: CanvasElement) => {
```
Comment (bryansamuel1) [inline src/tools/Ruler.ts]: Can we look at pulling out common functions from these tool files into its own folder ? It looks like this sendToWebsockets in particular is almost identical in every tool? Look at the other functions as well and see if any can sensibly be pulled out and turned into a common function?

File: src/tools/shapes/RulerCreator.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    layer.add(this.text)
+    layer.add(this.line)
+    return {
+      circle: this.circle,
+      line: this.line,
+      text: this.text
+    }
+  }
+
+  // eslint-disable-next-line
```
Comment (bryansamuel1) [inline src/tools/shapes/RulerCreator.ts]: what is eslint complaining about here btw?

File: src/tools/Circle.ts (hunk 1 above)
Comment (Eirmas) [reply src/tools/Circle.ts]: Will remove

File: src/tools/Ruler.ts (hunk 2 above)
Comment (Eirmas) [reply src/tools/Ruler.ts]: Good idea, will have a look at it

File: src/tools/shapes/RulerCreator.ts (hunk 3 above)
Comment (Eirmas) [reply src/tools/shapes/RulerCreator.ts]: Probably some old code from before the rewrite, will remove

**Comment by NikoGrano** [comment]
Remember also have some sleep 😂😅

**Pull request merged**